app/src/main/java/com/example/habitsmasher/NotificationFragment.java

```diff
@@ -0,0 +1,255 @@
+                                                _notificationItemAdapter._snapshots.get(position).getId());
+                                        resetOptionsOfAdapter();
+                                        break;
+                                }
+                            }
+                });
+        // connect listener to recycler view
+        recyclerView.addOnItemTouchListener(touchListener);
+    }
+
+    /**
+     * This readjusts the adapter of the RecyclerView so accepted/denied requests
+     * are deleted (as a consequence, when this occurs, any new requests
+     * sent after the user loads this fragment will also be loaded)
+     *
+     * Source:
+     * https://stackoverflow.com/questions/48014211/can-we-update-query-of-
+     * firestore-recycler-adapter-without-setting-the-whole-adap
+     */
+    protected void resetOptionsOfAdapter() {
+        _notificationItemAdapter.stopListening();
+        Query query = getListFromFirebase();
+        FirestoreRecyclerOptions<User> options = new FirestoreRecyclerOptions.Builder<User>()
+                                                                             .setQuery(query,
+                                                                                       User.class)
+                                                                             .build();
+        _notificationItemAdapter.updateOptions(options);
+        _notificationItemAdapter.startListening();
+    }
+
+    @Override
+    protected Query getListFromFirebase() {
+        DocumentReference sampleDocument = _db.collection("Users").document(_user.getId());
+        Task<DocumentSnapshot> querySnapshotTask = sampleDocument.get();
+
+        while (!querySnapshotTask.isComplete());
+
+        Map<String, Object> objectMap = querySnapshotTask.getResult().getData();
+
+        ArrayList<String> requestList = (ArrayList<String>) objectMap.get(FOLLOW_REQUEST_FIELD);
+        if(requestList != null && !requestList.isEmpty()) {
+            return _db.collection("Users").whereIn("id", requestList);
+        }
+
+        // return empty query when there are no follow requests
+        return _db.collection("Users").whereIn("id", Arrays.asList(""));
+    }
+
+    @Override
+    protected void populateList(Query query) {
+        // not needed
+    }
+
+    @Override
+    protected void openAddDialogBox() {
+        // not needed
+    };
+
+    @Override
+    public void openEditDialogBox(int position) {
+        // not needed
+    }
+
+    @Override
+    protected void openViewWindowForItem(int position) {
+        // not needed
+    }
+
+    @Override
+    public void updateListAfterAdd(User addedObject) {
+        // list does not support adding
+    };
+
+    @Override
+    public void updateListAfterEdit(User editedObject, int pos) {
+        // list does not support editing
+    };
+
+    @Override
+    public void updateListAfterDelete(int pos) {
+        // list does not support deletion
+    };
```

Comment:
Can we move all the unimplemented methods to the **bottom of this file**?

- [ ] move unimplemented methods to the bottom